### PR TITLE
ida-free: init at 8.1.0.221006

### PR DESCRIPTION
**Pull request by athre0z**
###### Description of changes

This PR adds a new package `ida-free` that packages up the freeware version of the proprietary [IDA Pro disassembler](https://hex-rays.com/ida-pro/). This is roughly inspired by [the corresponding AUR package](https://aur.archlinux.org/packages/ida-free).

###### Things done

- [ ] Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by bbjubjub2494 — APPROVED** [review]
Result of `nixpkgs-review pr 201544` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ida-free</li>
  </ul>
</details>

Works on my machine, LGTM

File: pkgs/applications/misc/ida-free/default.nix [hunk 1]
```diff
@@ -0,0 +1,105 @@
+{ lib, stdenv, fetchurl, libGL, zlib, libxkbcommon, glib, makeWrapper, xorg, dbus, fontconfig, freetype, makeDesktopItem }:
```
Comment (azahi) [inline pkgs/applications/misc/ida-free/default.nix:1]:
```suggestion
{ lib
, stdenv
, fetchurl
, libGL
, zlib
, libxkbcommon
, glib
, makeWrapper
, xorg
, dbus
, fontconfig
, freetype
, makeDesktopItem
}:
```

File: pkgs/applications/misc/ida-free/default.nix [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+with lib;
+
```
Comment (azahi) [inline pkgs/applications/misc/ida-free/default.nix:4]:
```suggestion
```

File: pkgs/applications/misc/ida-free/default.nix [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    libxkbcommon
+    dbus
+    fontconfig
+    freetype
+    libGL
+  ];
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  unpackPhase = ":";
```
Comment (azahi) [inline pkgs/applications/misc/ida-free/default.nix:63]:
```suggestion
  dontUnpack = true;
```

File: pkgs/applications/misc/ida-free/default.nix [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Freeware version of the world's smartest and most feature-full disassembler";
+    homepage = "https://hex-rays.com/ida-free/";
+    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
+    license = licenses.unfree;
+    maintainers = with maintainers; [ athre0z ];
+    platforms = [ "x86_64-linux" ];
```
Comment (azahi) [inline pkgs/applications/misc/ida-free/default.nix:103]:
```suggestion
    platforms = [ "x86_64-linux" ];
    mainProgram = "ida";
```

**Review by azahi — COMMENTED** [review]
Works and looks great. I just have a couple of nitpicks.

**Comment by athre0z** [comment]
Thanks for the feedback @azahi -- all good suggestions! I believe I have addressed all of them now.

**Review by azahi — APPROVED** [review]

**Comment by athre0z** [comment]
Hm, turns out that hex-rays occasionally swaps out the executable at the download path for patch updates. Since I can't just mirror it for licensing reasons, I'm closing this for the moment. If I find the energy to ask HR to grant me written mirroring permissions, I might re-open this at a later point.

**Comment by milahu** [comment]
> I can't just mirror it for licensing reasons

lets use the web archive for [ida-free](https://hex-rays.com/ida-free/)

[idafree82_linux.run](https://web.archive.org/web/*/https://out7.hex-rays.com/files/idafree82_linux.run)
[idafree82_mac.app.zip](https://web.archive.org/web/*/https://out7.hex-rays.com/files/idafree82_mac.app.zip)
[arm_idafree82_mac.app.zip](https://web.archive.org/web/*/https://out7.hex-rays.com/files/arm_idafree82_mac.app.zip)
[idafree82_windows.exe](https://web.archive.org/web/*/https://out7.hex-rays.com/files/idafree82_windows.exe)

**Comment by lourkeur** [comment]
> hmm. we can still create an `ida-free-installer` which will download ida-free on runtime

This is better addressed at build time with `requireFile` (e.g. https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/oraclejdk/jdk11-linux.nix#L14)

I have had a personal derivation for IDA free 8.2 for a while. Should I PR it or put it on NUR? https://github.com/lourkeur/distro/blob/main/pkgs/packages/ida-free.nix

**Pull request closed without merging**